src/rules/no-named-as-default-member.js

```diff
@@ -49,6 +49,8 @@ module.exports = function(context) {
   }
 
   function handleDestructuringAssignment(node) {
+    if (!node.init) return
+
```

Comment:
Nah, I'm fine with it. Style-wise I prefer mine, but clarity-wise yours has an explicit name to the condition, so both are good IMO.